pennylane/_qubit_device.py

```diff
@@ -477,24 +477,25 @@ def marginal_prob(self, prob, wires=None):
             return prob
 
         wires = Wires(wires)
-        # get indices of wires on the device's register
-        wire_indices = self.register.indices(wires)
-
         # determine which subsystems are to be summed over
-        inactive_registers = list(set(range(self.num_wires)) - set(wire_indices))
+        inactive_wires = Wires.unique_wires([self.wires, wires])
+
+        # translate to wire labels used by device
+        wires = self.map_wires(wires)
+        inactive_wires = self.map_wires(inactive_wires)
 
         # reshape the probability so that each axis corresponds to a wire
         prob = self._reshape(prob, [2] * self.num_wires)
 
         # sum over all inactive wires
-        prob = self._flatten(self._reduce_sum(prob, inactive_registers))
+        prob = self._flatten(self._reduce_sum(prob, inactive_wires.labels))
```

Comment:
The "label" attribute makes a lot of code more readable!

Reply:
What does `wire.labels` return in this case? I assumed it would return the users labels, which doesn't make sense in this context 🤔 

Ohhh... it returns the devices wire labels, which in this case are ints?

Reply:
Will this be an issue if the device uses strings for its labels? I suppose a lot of things will break in that case

Reply:
Yes, the device would have to rewrite this function.

I'll follow @co9olguy 's recommendation to name the translated (device) wires differently, then the confusion should be resolved.

Reply:
@antalszava, since this may affect the device refactor: Good to be aware that at the moment, `QubitDevice` is coded to only work for devices with integer-valued internal wire labels...Maybe something to try and change in future?

Reply:
Hi @mariaschuld, thank you for the heads-up! :slightly_smiling_face:

For my own understanding: would that mean that `QubitDevice`
* can internally handle `Wires` instances so long as they contain integer-valued wire labels and
* it would be a to-do to allow for example string typed labels too?
(still getting up to speed with the related changes)

Reply:
Yes exactly, but IMO not an urgency, as all our qubit devices use integer-labelled wires...just making you aware if it happens to be trivial to change...